File: Youtube_java/user/Comments.java
package user;
import java.time.LocalDate;
import java.io.Serializable;
import idGenerator.*;

public class Comments implements Serializable {
    private static final long serialVersionUID = 1L;
    
    public String commentId;
    public LocalDate date;
    public String userName;
    public String text;
    public int likes;

    public Comments() {
        this.commentId = IdGenerator.generateCommentId();
        this.date = LocalDate.now();
        this.likes = 0;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public int getLikes() {
        return likes;
    }

    public void setLikes(int likes) {
        this.likes = likes;
    }

    public String getCommentId() {
        return commentId;
    }

    public void setCommentId(String commentId) {
        this.commentId = commentId;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    // Créer un commentaire avec ID auto-généré
    public static Comments createComment(String name, LocalDate date, String text) {
        Comments comments = new Comments();
        comments.date = date;
        comments.userName = name;
        comments.text = text;
        return comments;
    }

    // Afficher un commentaire
    public void displayComment() {
        System.out.println("\t \t Commentaire :" + getUserName() + "\n");
        System.out.println("Id du commentaire:" + getCommentId() + "\nDate de publication :" + getDate() + "\nTexte: " + getText() + "\nLikes :" + getLikes());
    }
    
    // Convertir en chaîne pour sauvegarde
    public String serialize() {
        return commentId + "|" + userName + "|" + text + "|" + likes + "|" + date.toString();
    }
    
    // Créer à partir d'une chaîne
    public static Comments deserialize(String data) {
        String[] parts = data.split("\\|");
        Comments comment = new Comments();
        comment.commentId = parts[0];
        comment.userName = parts[1];
        comment.text = parts[2];
        comment.likes = Integer.parseInt(parts[3]);
        comment.date = LocalDate.parse(parts[4]);
        return comment;
    }
}

File: Youtube_python/comments.py
from datetime import date
from id_generator import IdGenerator

class Comments:
    def __init__(self):
        self.comment_id = IdGenerator.generate_comment_id()
        self.date = date.today()
        self.user_name = ""
        self.text = ""
        self.likes = 0

    @staticmethod
    def create_comment(name, date_pub, text):
        comment = Comments()
        comment.date = date_pub
        comment.user_name = name
        comment.text = text
        return comment

    def display_comment(self):
        print(f"\t \t Commentaire : {self.user_name}\n")
        print(f"Id: {self.comment_id}\nDate: {self.date}\nTexte: {self.text}\nLikes: {self.likes}")

    def serialize(self):
        return f"{self.comment_id}|{self.user_name}|{self.text}|{self.likes}|{self.date.isoformat()}"

    @staticmethod
    def deserialize(data):
        parts = data.split("|")
        comment = Comments()
        comment.comment_id = parts[0]
        comment.user_name = parts[1]
        comment.text = parts[2]
        comment.likes = int(parts[3])
        comment.date = date.fromisoformat(parts[4])
        return comment
